### DISCUSSION TITLE
Need help with scaling pocketbase in production (advice)

### DISCUSSION THREAD

**Discussion by Jaydeep189** [pocketbase/pocketbase#5586] [Q&A]
Hi, I am using pocketbase in production for quite a while and it has been a amazing experience.

It's a data heavy platform, many data related to the user is stored based on which service they are using

My client has come by and said that he needs to start white labeling his platform, so this means that we would scale up in requests as now many users will be interacting with the platform, and the data related to the user will also scale more rapidly then before anticipated.

Before implementing the white label solution, i just needed to know how to scale pocketbase in this situation (currently using 8 GB ram, 4 core CPU server).

Just wanted to know how do we scale, do we add more instances (if that's possible?), or do we scale vertically (will my CPU bottleneck in this case?)

Also wanted to know what are the limitations?

PS: I am not a DevOps expert, so please correct me if i have written something stupid.

**Comment by Jaydeep189** [comment]
Just for a little context, current the platform doesn't have that many request.

![image](https://github.com/user-attachments/assets/1afe9832-c9de-460e-a758-f7c8748234a1)

There are sudden bursts for request as seen in the screenshot, that are for reports being imported (where many records are created in more then one collection)

All the other spikes are from normal users using the service on the platform, currently slightly above (> 200 users) and 1-2 user on-boarding each day organically (not marketed the platform yet)

**Comment by lumosminima** [comment]
You haven't shared where you are actually encountering scaling problems at present _(log screenshot doesn't show: level = 8_). You can't "scale" something if you haven't benchmarked and don't know where the current bottlenecks are. Is disk filling up? throw more storage at it or offload files to S3. High memory usage? maybe an app level memory leak or an actual memory bottleneck. In any case you can only scale pocketbase vertically for now see: [https://pocketbase.io/faq/#:~:text=does%20it%20scale?](https://pocketbase.io/faq/#:~:text=does%20it%20scale?) there are also community projects like #487 that allow multiple instances but pocketbase hooks and realtime api don't work with these, as those work at app level and projects like the one linked do db replication.

Past discussions on same topic:
#2184
#2390
#1673
#395
#615
#515
